modules/Material/Dialog.qml

```diff
@@ -19,118 +19,186 @@
+                        placeholder.anchors.leftMargin +
+                        placeholder.anchors.rightMargin))
+
+        height: titleLabel.height +
+                titleLabel.anchors.topMargin +
+                placeholder.childrenRect.height +
+                placeholder.anchors.topMargin +
+                placeholder.anchors.bottomMargin +
+                btnContainer.height
+
+        Label {
+            id: titleLabel
+            height: implicitHeight
+            style: "title"
+            anchors {
+                top: parent.top
+                topMargin: units.dp(16)
+                left: parent.left
+                leftMargin: units.dp(16)
+            }
+        }
+
+        Item {
+            z: parent.z + 5
+            anchors {
+                left: parent.left
+                leftMargin: units.dp(16)
+                right: parent.right
+                rightMargin: units.dp(16)
+                top: titleLabel.bottom
+                topMargin: units.dp(16)
+                bottom: btnContainer.top
+                bottomMargin: hasActions ? 0 : units.dp(16)
+            }
+            id: placeholder
+        }
+
+        Item {
+            id: btnContainer
+            z: parent.z + 5
+            width: negativeBtn.width + positiveBtn.width + units.dp(24)
+            height: hasActions ? units.dp(64) : 0
+            visible: hasActions
+
+            anchors {
+                bottom: parent.bottom
+                right: parent.right
+            }
+
+            Button {
+                id: negativeBtn
+
+                text: negativeBtnText
+                textColor: Theme.accentColor
+                anchors {
+                    top: parent.top
+                    topMargin: units.dp(8)
+                    right: positiveBtn.left
+                    rightMargin: units.dp(8)
+                    bottom: parent.bottom
+                    bottomMargin: units.dp(8)
+                }
+                onClicked: {
+                    close();
+                    rejected();
+                }
+            }
+
+            Button {
+                id: positiveBtn
+
+                text: positiveBtnText
+                textColor: Theme.accentColor
+                anchors {
+                    top: parent.top
+                    topMargin: units.dp(8)
+                    right: parent.right
+                    rightMargin: units.dp(16)
+                    bottom: parent.bottom
+                    bottomMargin: units.dp(8)
+                }
+                onClicked: {
+                    close()
+                    accepted();
+                }
+            }
+        }
+
+        MouseArea {
+            anchors.fill: parent
+            propagateComposedEvents: false
+            onClicked: {
+                mouse.accepted = false
+            }
+        }
+    }
+
+    function show() {
+        open(Utils.findRoot(parent))
```

Comment:
i haven't tested this but doesn't this work with just parent since you are setting it in Popover?